src/http/server/handlers/log_handler.cr

```diff
@@ -1,20 +1,27 @@
+require "log"
+
 # A handler that logs the request method, resource, status code, and
-# the time used to execute the next handler, to the given `IO`.
+# the time used to execute the next handler, to the log with source `http.server`.
 class HTTP::LogHandler
   include HTTP::Handler
+  Log = ::Log.for("http.server")
+
+  @[Deprecated("Use `new` without arguments instead")]
+  def initialize(io : IO)
+  end
 
-  # Initializes this handler to log to the given `IO`.
-  def initialize(@io : IO = STDOUT)
+  def initialize
   end
 
   def call(context)
     elapsed = Time.measure { call_next(context) }
     elapsed_text = elapsed_text(elapsed)
 
-    @io.puts "#{context.request.method} #{context.request.resource} - #{context.response.status_code} (#{elapsed_text})"
+    Log.info { "#{context.request.method} #{context.request.resource} - #{context.response.status_code} (#{elapsed_text})" }
   rescue e
-    @io.puts "#{context.request.method} #{context.request.resource} - Unhandled exception:"
-    e.inspect_with_backtrace(@io)
+    unless context.response.closed?
+      Log.error(exception: e) { "#{context.request.method} #{context.request.resource} - Unhandled exception:" }
```

Comment:
Slightly different story but also worth fixing: In case `Log.info` or `elapsed_text` fails, this `rescue` block would be triggered. I don't think that's expected because it's not an error related to the HTTP application, but the internal logging setup. The `rescue` should just cover `call_next(context)`, not even `Time.measure` (which is also an internal detail and failing that is not relevant to the HTTP protocol).

Reply:
Agreed. I also noticed that but I was trying to focus just on the logging and not handling client disconnections for this PR.
I'll make the change anyway.

Reply:
Ok, solved on this PR